src/webgpu/api/validation/encoding/cmds/compute_pass.spec.ts

```diff
@@ -65,7 +76,17 @@ setPipeline should generate an error iff using an 'invalid' pipeline.
 g.test('pipeline,device_mismatch')
   .desc('Tests setPipeline cannot be called with a compute pipeline created from another device')
   .paramsSubcasesOnly(u => u.combine('mismatched', [true, false]))
-  .unimplemented();
+  .fn(async t => {
+    const { mismatched } = t.params;
+
+    const pipeline = mismatched
+      ? t.createComputePipelineForMismatch(t.mismatchedDevice)
+      : t.createComputePipelineForMismatch(t.device);
```

Comment:
nit: Since createComputePipelineForMismatch is only used in this one place, I don't think it needs to be in a shared location. I would either define it as a function inside this function, or just do it inline like

```
const pipelineDevice = mismatched ? t.mismatchedDevice : t.device;
const pipeline = pipelineDevice.createComputePipeline( // ...
```